src/transformers/models/whisper/tokenization_whisper.py

```diff
@@ -705,3 +705,272 @@ def get_decoder_prompt_ids(self, task=None, language=None, no_timestamps=True):
         forced_tokens = self.prefix_tokens[1:]
         forced_decoder_ids = [(rank + 1, token) for rank, token in enumerate(forced_tokens)]
         return forced_decoder_ids
+
+    def _decode_asr(self, model_outputs, *, return_timestamps, return_language, time_precision):
+        """
+        Internal method meant to only be used by asr pipeline. Handles all the little quirks specific to whisper to
+        handle the various options not allowed in other seq2seq models
+        """
+
+        # =========== Overview ============
+        # - iterate over all outputs
+        # - all tokens within output
+        # - Each token can be
+        #   - language token
+        #   - special token
+        #   - timestamp token
+        #   - text token
+        # - We accumulate the text tokens.
+        # - We split on end timestamps
+        # - Lots of complexity comes from stride and timestamps
+
+        last_language = None
+
+        def new_chunk():
+            return {"language": last_language, "timestamp": [None, None], "text": ""}
+
+        # Welcome to the state machine !
+        chunks = []
+        chunk = new_chunk()
+        time_offset = 0.0
+        timestamp_begin = self.convert_tokens_to_ids("<|notimestamps|>") + 1
+        previous_tokens = []
+        skip = False
+        right_stride_start = None
+
+        all_special_ids = set(self.all_special_ids)
+        all_special_ids.add(50256)
```

Comment:
Why? A comment would be appreciated.

Reply:
This is because `pad_token_id` is sometimes set as 50256, which is a regular token, not `all_special_ids`.

This leads to weird conflict handling (since those ids get interpreted as real text, hence depend on batch size). 

This is imo an issue with the config of the models. However that was done on purpose iiuc. @ArthurZucker do you remember/have a link with the original issue ?

I removed this because it doesn't belong here regardless.